src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -340,3 +345,94 @@ def load_secure_props(self) -> None:
                 f" with error '{error_msg}'",
                 SecurePropsNotFoundWarning,
             )
+
+    def _retrieve_credential(self, service_name: str) -> Optional[str]:
+        """
+        Retrieve the credential from the keyring or storage.
+        If the credential exceeds the maximum length, retrieve it in parts.
+        Parameters
+        ----------
+        service_name: str
+            The service name for the credential retrieval
+        Returns
+        -------
+        str
+            The retrieved  encoded credential
+        """
+        encoded_credential = keyring.get_password(service_name, constants["ZoweAccountName"])
+
+        if encoded_credential is None:
+            # Retrieve the secure value with an index
+            index = 1
+            temp_value = keyring.get_password(f"{service_name}-{index}", f"{constants['ZoweAccountName']}-{index}")
+            while temp_value is not None:
+                if encoded_credential is None:
+                    encoded_credential = temp_value
+                else:
+                    encoded_credential += temp_value
+                index += 1
+                temp_value = keyring.get_password(f"{service_name}-{index}", f"{constants['ZoweAccountName']}-{index}")
+                
+        if encoded_credential is not None and encoded_credential.endswith("\0"):
+            encoded_credential = encoded_credential[:-1]
+            
+        
+        return encoded_credential 
+    
+    def set_secure_props(self) -> None:
+        """
+        Set secure_props for the given config file
+        Returns
+        -------
+        None
+        """
+        if not HAS_KEYRING:
+            return
+
+        try:
+            service_name = constants["ZoweServiceName"]
+            credential = self.secure_props.get(self.filepath, {})
```

Comment:
We want to store the entire credentials object which has file paths as keys:
`credential = {self.filepath: self.secure_props}`